app/models/arm.rb

```diff
@@ -48,6 +48,14 @@ class Arm < ActiveRecord::Base
   validates :visit_count, numericality: { greater_than: 0 }
   validates :subject_count, numericality: { greater_than: 0 }
 
+  validate do |arm|
```

Comment:
I agree, but I couldn't figure out another way to make the validations chain back down to the visit groups. Let's think on it and try to refactor down the road.